src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -38,6 +64,92 @@ public bool HasValue
         }
 
         /// <summary>
+        /// Returns the value of the domain part of the host. The port is removed if it was present.
+        /// IPv6 addresses will have brackets added if they are missing.
+        /// </summary>
+        /// <returns></returns>
+        public string Domain
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_value))
+                {
+                    return string.Empty;
+                }
+
+                // normalize the value
+                var uriComponent = ToUriComponent();
```

Comment:
ToUriComponent returns the punycode form. We'd rather return the unicode form from this property. That's the form you'd get from Value and pass into the constructor.